Firestore/Source/Local/FSTLocalStore.mm

```diff
@@ -213,6 +215,7 @@ - (FSTLocalWriteResult *)locallyWriteMutations:(NSArray<FSTMutation *> *)mutatio
 
 - (FSTMaybeDocumentDictionary *)acknowledgeBatchWithResult:(FSTMutationBatchResult *)batchResult {
   return self.persistence.run("Acknowledge batch", [&]() -> FSTMaybeDocumentDictionary * {
+    FSTListenSequenceNumber sequenceNumber = [self.listenSequence next];
```

Comment:
All the sequence number plumbing in here seems a little gross, especially since we sequence these things for GC

Is there some way to make the sequence number a property of the garbage collector/reference delegate?